build.gradle

```diff
@@ -46,6 +47,29 @@ ext.compatibilityVersion = '1.7'
 sourceCompatibility = compatibilityVersion
 targetCompatibility = compatibilityVersion
 
+sourceSets {
+    main.java.srcDirs = ['src/main/java'
+                        ,'src/generated/java' ]
+}
+
+compileJava.doFirst {
+ File generatedDir = file("${rootDir}/src/generated/java")
+ if (generatedDir.exists()) {
+  delete generatedDir
+ }
+ generatedDir.mkdirs()
+}
+
+clean.doFirst {
+ File generatedDir = file("${rootDir}/src/generated/java")
+ if (generatedDir.exists()) {
+  delete generatedDir
+ }
+ generatedDir.mkdirs()
+}
```

Comment:
This would then get generated for all users and use cases and put the `generated` folder under `src` which I don't think is something we need?